packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/ui/viewModel/ConnectionStateViewModel.kt

```diff
@@ -284,6 +289,52 @@ class ConnectionStateViewModel(
     }
 }
 
+/**
+ * When we modify the target cluster somehow through DataGrip, we might want to retrigger the
+ * analysis of the scope.
+ */
+class DataSourcesChangesAuditor : DataAuditor {
+    @VisibleForTesting
+    internal var readContext: (Context) -> Triple<LocalDataSource, Project, String>? = { context ->
+        val dataSource = context.connection?.connectionPoint?.dataSource
+        val project = dataSource?.findProjectWhereActive(null)
+        val query = context.query
+
+        if (dataSource == null || project == null || query == null) {
+            null
+        } else {
+            Triple(dataSource, project, query)
+        }
+    }
+
+    override fun requestFinished(context: Context) {
+        val (dataSource, project, query) = readContext(context) ?: return
+
+        if (likelyChangesDataSource(query)) {
+            dataSource.incModificationCount()
+
+            val codeEditorViewModel by project.service<AnalysisScopeViewModel>()
+            runBlocking {
+                codeEditorViewModel.reanalyzeCurrentScope()
```

Comment:
```suggestion
                codeEditorViewModel.reanalyzeRelevantEditors()
```
I did not find anything such as reanalyzeCurrentScope. Maybe a mistype?

Reply:
Ah yes, the typo is actually in the variable name, the class is correct. Fixing it in a new commit. Likely missed the variable name while I was refactoring.